crates/sui-core/src/checkpoints/mod.rs

```diff
@@ -66,6 +66,9 @@ pub struct CheckpointStore {
     /// Maps sequence number to checkpoint summary
     checkpoint_summary: DBMap<CheckpointSequenceNumber, CheckpointSummary>,
 
+    /// How many txns were in a given checkpoint?
+    checkpoint_size: DBMap<CheckpointSequenceNumber, u64>,
```

Comment:
+1 You can refer to how we add the gas cost in the checkpoint summary

Reply:
The previous CheckpointSummary does not have enough info. We would have to load and deserialize the previous contents in order to determine their size. Some ideas:

- We could track the previous size in memory and initialize it by loading the last contents at startup.
- We could add the number of transactions to CheckpointSummary, so that it would have enough information to compute the next starting sequence.

Reply:
We could store the seq number of the last transaction in the checkpoint? Or alternatively the seq number of the first transaction in the next checkpoint.

Reply:
Ok, fixed this by adding a `num_transactions` field to CheckpointSummary which seems like a generally useful thing to have anyway.

Reply:
What I meant is that we only need to store the seq number of the last transaction in the checkpoint, instead of 2 numbers. That's the most efficient and reduces 8 bytes from the summary.

Reply:
Oh, I thought you were suggesting we also store the final sequence number in addition to the first. This is equally efficient to my original implementation, and lets us avoid having another database table, but seems unnecessarily awkward. If we really want to save those 8 bytes in the summary I would rather do the extra work of keeping the previous checkpoint size in memory.

But is it really worth saving 8 bytes here? The extra number is redundant, but makes things so much simpler.

Reply:
(we could probably also use a u32 here).

Reply:
agree on " seems unnecessarily awkward." Also TBH I'm not sure how the # of tx in a checkpoint is going to be helpful. So maybe start with `do the extra work of keeping the previous checkpoint size in memory.`